### ISSUE TITLE
Email editor is broken

### ISSUE THREAD

**Issue by comunidademagento** [mautic/mautic#7284]
[//]: # ( Invisible comment: 
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Before you create the issue:
IIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIIII
Search for similar report among other reported issues.
Learn how to troubleshoot at https://www.mautic.org/docs/en/tips/troubleshooting.html
Use drag&drop to attach images or other files )

## Bug Description
We try select email and open in email editor

| Q   | A
| --- | ---
| Mautic version |  2.15.0
| PHP version | 7.0
| Browser | Google Chrome

### Steps to reproduce
1. We try select email and open in email editor
 
### Log errors

VM264:2 Uncaught SyntaxError: Unexpected token ;
    at Object.Mautic.updateIframeContent (app.js?vd7677c99:162)
    at Object.Mautic.buildBuilderIframe (app.js?vd7677c99:159)
    at Object.Mautic.cloneHtmlContent (app.js?vd7677c99:165)
    at Object.Mautic.sendBuilderContentToTextarea (app.js?vd7677c99:163)
    at HTMLButtonElement.<anonymous> (app.js?vd7677c99:150)
    at HTMLButtonElement.dispatch (libraries.js?vd7677c99:404)
    at HTMLButtonElement.elemData.handle (libraries.js?vd7677c99:379)
Mautic.updateIframeContent @ app.js?vd7677c99:162
Mautic.buildBuilderIframe @ app.js?vd7677c99:159
Mautic.cloneHtmlContent @ app.js?vd7677c99:165
Mautic.sendBuilderContentToTextarea @ app.js?vd7677c99:163
(anonymous) @ app.js?vd7677c99:150
dispatch @ libraries.js?vd7677c99:404
elemData.handle @ libraries.js?vd7677c99:379
app.js?vd7677c99:11 Uncaught ReferenceError: mauticLang is not defined
    at Object.translate (app.js?vd7677c99:11)
    at app.js?vd7677c99:349
    at Object.Mautic.updateIframeContent (app.js?vd7677c99:162)
    at Object.Mautic.buildBuilderIframe (app.js?vd7677c99:159)
    at Object.Mautic.cloneHtmlContent (app.js?vd7677c99:165)
    at Object.Mautic.sendBuilderContentToTextarea (app.js?vd7677c99:163)
    at HTMLButtonElement.<anonymous> (app.js?vd7677c99:150)
    at HTMLButtonElement.dispatch (libraries.js?vd7677c99:404)
    at HTMLButtonElement.elemData.handle (libraries.js?vd7677c99:379)
translate @ app.js?vd7677c99:11
(anonymous) @ app.js?vd7677c99:349
Mautic.updateIframeContent @ app.js?vd7677c99:162
Mautic.buildBuilderIframe @ app.js?vd7677c99:159
Mautic.cloneHtmlContent @ app.js?vd7677c99:165
Mautic.sendBuilderContentToTextarea @ app.js?vd7677c99:163
(anonymous) @ app.js?vd7677c99:150
dispatch @ libraries.js?vd7677c99:404
elemData.handle @ libraries.js?vd7677c99:379
app.js?vd7677c99:7 Uncaught ReferenceError: mauticAjaxUrl is not defined
    at HTMLDocument.<anonymous> (app.js?vd7677c99:7)
    at fire (libraries.js?vd7677c99:328)
    at Object.fireWith [as resolveWith] (libraries.js?vd7677c99:333)
    at Function.ready (libraries.js?vd7677c99:340)
    at HTMLDocument.completed (libraries.js?vd7677c99:340)
    at Object.Mautic.updateIframeContent (app.js?vd7677c99:162)
    at Object.Mautic.buildBuilderIframe (app.js?vd7677c99:159)
    at Object.Mautic.cloneHtmlContent (app.js?vd7677c99:165)
    at Object.Mautic.sendBuilderContentToTextarea (app.js?vd7677c99:163)
    at HTMLButtonElement.<anonymous> (app.js?vd7677c99:150)
(anonymous) @ app.js?vd7677c99:7
fire @ libraries.js?vd7677c99:328
fireWith @ libraries.js?vd7677c99:333
ready @ libraries.js?vd7677c99:340
completed @ libraries.js?vd7677c99:340
Mautic.updateIframeContent @ app.js?vd7677c99:162
Mautic.buildBuilderIframe @ app.js?vd7677c99:159
Mautic.cloneHtmlContent @ app.js?vd7677c99:165
Mautic.sendBuilderContentToTextarea @ app.js?vd7677c99:163
(anonymous) @ app.js?vd7677c99:150
dispatch @ libraries.js?vd7677c99:404
elemData.handle @ libraries.js?vd7677c99:379
app.js?vd7677c99:167 Uncaught TypeError: Cannot read property 'find' of undefined
    at Object.Mautic.destroySlots (app.js?vd7677c99:167)
    at app.js?vd7677c99:150
    at app.js?vd7677c99:163
    at app.js?vd7677c99:165
    at HTMLIFrameElement.<anonymous> (app.js?vd7677c99:159)
    at HTMLIFrameElement.dispatch (libraries.js?vd7677c99:404)
    at HTMLIFrameElement.elemData.handle (libraries.js?vd7677c99:379)

[//]: # ( Invisible comment:
Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable. )

![img01](https://user-images.githubusercontent.com/7476455/53416524-72882e00-39b2-11e9-97de-824b33823e86.png)

**Comment by comunidademagento** [comment]
Hello,

See more informations in this video.

https://www.useloom.com/share/f815b53a4c6c439db33c30e5e204c713

Thanks!!!

**Comment by comunidademagento** [comment]
Hello,

See more informations ...

https://www.useloom.com/share/4ff408371f2d4aa9b51b6a27ff84a071

SyntaxError: missing ) after for-loop control new:2:283

    updateIframeContent https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:162 buildBuilderIframe https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:159 prepareBuilderIframe https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:227 initBuilderIframe https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:227 launchBuilder https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:151 onclick https://marketing.comunidademagento.com.br/s/emails/new:1 trigger https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:400 trigger https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:437 each https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:149 each https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:139 trigger https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:437 name https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:602 buttonClick https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:67 dispatch https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:404 handle https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:379 

TypeError: "document.getElementById(...).contentWindow.Mautic is undefined"

    launchBuilder https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:150 sendBuilderContentToTextarea https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:163 launchBuilder https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:150 dispatch https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:404 handle https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:379 

app.js:163:866

    sendBuilderContentToTextarea https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:163 launchBuilder https://marketing.comunidademagento.com.br/media/js/app.js?vd7677c99:150 dispatch https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:404 handle https://marketing.comunidademagento.com.br/media/js/libraries.js?vd7677c99:379

**Comment by npracht** [comment]
Do you have something in mautic logs ?

**Comment by comunidademagento** [comment]
Hello @npracht 

See logs - https://marketing.comunidademagento.com.br/logs_2019.03.07_18.32.39.tgz

See images - https://imgur.com/a/h45xQ5k

=============== APP/LOGS

[2019-03-07 01:10:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []
[2019-03-07 01:30:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 01:35:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:rebuild` [] []
[2019-03-07 01:40:22] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []
[2019-03-07 02:01:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 02:05:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:rebuild` [] []
[2019-03-07 02:30:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 02:40:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []
[2019-03-07 03:01:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 14:40:12] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []

=============== CONSOLE BROWSER

app.js?vd7677c99:163 TypeError: Cannot read property 'destroySlots' of undefined
    at app.js?vd7677c99:150
    at Object.Mautic.sendBuilderContentToTextarea (app.js?vd7677c99:163)
    at HTMLButtonElement.<anonymous> (app.js?vd7677c99:150)
    at HTMLButtonElement.dispatch (libraries.js?vd7677c99:404)
    at HTMLButtonElement.elemData.handle (libraries.js?vd7677c99:379)

=============== SYSTEM INFO

[2019-03-07 01:01:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 01:05:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:rebuild` [] []
[2019-03-07 01:10:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []
[2019-03-07 01:30:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 01:35:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:rebuild` [] []
[2019-03-07 01:40:22] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] Connection refused (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []
[2019-03-07 02:01:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 02:05:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:rebuild` [] []
[2019-03-07 02:30:03] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 02:40:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []
[2019-03-07 03:01:02] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:segments:update` [] []
[2019-03-07 14:40:12] mautic.NOTICE: Doctrine\DBAL\Exception\ConnectionException: An exception occured in driver: SQLSTATE[HY000] [2002] No such file or directory (uncaught exception) at /home/159509.cloudwaysapps.com/tqqdgwcaxq/public_html/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/AbstractMySQLDriver.php line 103 while running console command `mautic:campaigns:trigger` [] []

**Comment by glaubersilva** [comment]
I'm going through exactly the same problem. If I create a new email and then click "Save & Close" or "Apply" without entering the email builder I can no longer edit anything when I re-open the email, I can only edit the e -mail if I change the template. If I try to edit directly without changing the template the constructor simply does not allow editing anything and still hangs when I try to close it.

But when I create a new email and enter the builder before clicking "Save & Close" or "Apply" the problem does not happen and the errors in the console do not appear.

In short, one way to get around the problem while the bug is not fixed is to enter the constructor before clicking "Save & Close" or "Apply."

**Comment by Mancy** [comment]
I can re-producing the error:
while in the code editor window do this command in the JS console
```
mQuery(Mautic.builderCodeMirror.getValue())
```
or
```
mQuery("{subject}");
```

**Comment by RCheesley** [comment]
@comunidademagento @Mancy @glaubersilva are you still experiencing this problem with 2.15.3? If so please respond below, otherwise I'll close this down.

**Comment by coutogomes** [comment]
I update to 2.16.0 and still show the same error.

![Captura de tela 2020-02-20 15 23 16](https://user-images.githubusercontent.com/38143814/74966083-2caf4b80-53f5-11ea-83b8-26aa3d0dfbcf.png)

**Comment by madhuralandge** [comment]
Hello, 
I am using mautic v2.16.2.  I am not able to edit an email template.

Uncaught Error: History.js Adapter has already been loaded...
    at libraries.js?v46598394:733
    at libraries.js?v46598394:734
3143647:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
apu.php:1 Failed to load resource: net::ERR_BLOCKED_BY_CLIENT
app.js?v46598394:188

Uncaught TypeError: mQuery(...).sortable is not a function
    at Object.Mautic.initSections (app.js?v46598394:188)
    at HTMLDocument.<anonymous> (app.js?v46598394:243)
    at fire (libraries.js?v46598394:328)
    at Object.fireWith [as resolveWith] (libraries.js?v46598394:333)
    at Function.ready (libraries.js?v46598394:340)
    at HTMLDocument.completed (libraries.js?v46598394:340)
app.js?v46598394:110 

Uncaught TypeError: mQuery(...).chosen is not a function
    at Object.Mautic.activateChosenSelect (app.js?v46598394:110)
    at HTMLDocument.Mautic.dynamicContentOnLoad (app.js?v46598394:494) 
Getting these errors.

I have tried these commands also
php app/console cache:clear
php app/console cache:warmup
php app/console mautic:assets:generate

**Comment by RCheesley** [comment]
Hi there, as we are only providing fixes for Mautic 3.x going forward, can you confirm if this continues in Mautic 3.0.1?
Thanks!

**Comment by npracht** [comment]
Hi, without any feedback we cannot carry this issue open anymore. I'll close it and in case a new comment justify to re-open it, I'll be glad to do it :)

**Comment by jarridlima** [comment]
Hi,

I'm starting with Mautic 3.1.0 (via AWS EC2 Bitnami) and I receive this error on try create a e-mail by constructor:
![image](https://user-images.githubusercontent.com/8194683/92416300-313e1600-f133-11ea-972b-d041f58055ef.png)

Uncaught SyntaxError: Unexpected token u in JSON at position 0
    at JSON.parse (<anonymous>)
    at Object.Mautic.initGrapesJS (app.js?v1586e587:706)
    at Object.Mautic.launchBuilder (app.js?v1586e587:706)
    at HTMLButtonElement.onclick (emails:1)
    at Object.trigger (libraries.js?v1586e587:683)
    at HTMLButtonElement.<anonymous> (libraries.js?v1586e587:688)
    at Function.each (libraries.js?v1586e587:153)
    at jQuery.fn.init.each (libraries.js?v1586e587:144)
    at jQuery.fn.init.trigger (libraries.js?v1586e587:688)
    at jQuery.fn.init.jQuery.fn.<computed> [as click] (libraries.js?v1586e587:796)

None interface buttons works.

**Comment by jarridlima** [comment]
Hi, this worked for me:
---------------------
1 - Open Putty (or connection via SSH), navigate to Mautic directory like this:
cd /var/www/html/mautic 
OR
cd /opt/bitnami/apps/mautic/htdocs (in my case)

2 - Then run this command:
sudo php bin/console mautic:assets:generate

3 - Clear your browser cache, done!

Solution by @hassancs91 in https://github.com/mautic/mautic/issues/9171#issuecomment-687605469
